server/service/src/rnr_form/insert.rs

```diff
@@ -170,13 +170,19 @@ fn validate(
             // this should never happen, we've already checked it's there
             .ok_or(InsertRnRFormError::PeriodNotInProgramSchedule)?;
 
-        if previous_period != this_period + 1 {
-            return Err(InsertRnRFormError::PeriodNotNextInSequence);
+        // Periods are ordered by end_date desc in get_schedules_with_periods_by_program, so condition operator implies opposite to what error states
```

Comment:
I can't comment on above but position is causing an overflow... so I can't create another form after my initial form creation... haven't reviewed past this.